nixos/modules/services/web-apps/linkding.nix

```diff
@@ -0,0 +1,41 @@
+{
+  config,
+  lib,
+  options,
+  pkgs,
+  ...
+}:
+{
+  options.services.linkding = {
+    dataDir = lib.mkOption {
+      type = lib.types.str;
+      default = "/etc/linkding/data";
+      description = "The directory where linkding stores its data files.";
+    };
+
+    enable = lib.mkEnableOption "linkding service";
+
+    package = lib.mkPackageOption pkgs "linkding" { };
+
+    port = lib.mkOption {
+      default = 9090;
+      description = "Port on which linkding will listen for connections.";
```

Comment:
I think a host option would also be useful. Values could for example be "127.0.0.1" (only available on localhost), "0.0.0.0" (available elsewhere)